src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -1720,11 +1720,12 @@ async function renameWithQuickPick(c: ITerminalServicesCollection, accessor: Ser
 	}
 
 	if (instance) {
+		const originalTitle = instance.title;
 		const title = await accessor.get(IQuickInputService).input({
 			value: instance.title,
 			prompt: localize('workbench.action.terminal.rename.prompt', "Enter terminal name"),
 		});
-		instance.rename(title);
+		instance.rename(title ?? originalTitle);
```

Comment:
We just want to make sure we're not redoing it, so should we just guard with `if (title)`?

Reply:
Sure